kolibri/core/auth/tasks.py

```diff
@@ -676,3 +678,28 @@ def cleanupsync(**kwargs):
 
     sync_filter = kwargs.pop("sync_filter")
     call_command("cleanupsyncs", sync_filter=str(sync_filter), expiration=1, **kwargs)
+
+
+@register_task(
+    job_id="cleanup_expired_deleted_users",
+    queue=facility_task_queue,
+)
+def cleanup_expired_deleted_users():
+    """
+    Delete FacilityUsers whose date_deleted is more than 30 days ago.
+    If any soft-deleted users remain, re-enqueue this task to run again in 24 hours.
+    """
+    now = timezone.now()
+    threshold = now - datetime.timedelta(days=30)
+    expired_users = FacilityUser.soft_deleted_objects.filter(
+        date_deleted__lte=threshold
+    )
+    expired_users_count = expired_users.count()
```

Comment:
Why do a count check here? Can just call delete regardless, surely?

Reply:
I don't know why I added it there 😅.